cpp/src/plasma/lib/java/org_apache_arrow_plasma_PlasmaClientJNI.cc

```diff
@@ -102,13 +102,13 @@ JNIEXPORT jobject JNICALL Java_org_apache_arrow_plasma_PlasmaClientJNI_create(
   std::shared_ptr<Buffer> data;
   Status s = client->Create(oid, size, md, md_size, &data);
   if (s.IsPlasmaObjectExists()) {
-    jclass Exception = env->FindClass("java/lang/Exception");
+    jclass Exception = env->FindClass("org/apache/arrow/plasma/exceptions/DuplicateObjectException");
     env->ThrowNew(Exception,
                   "An object with this ID already exists in the plasma store.");
```

Comment:
no need to pass this message `"An object with this ID already exists in the plasma store."` to the exception. because the exception type already indicates this. 
Instead, it'd be useful to pass the object id, so that we can know which object is duplicate